src/frequenz/sdk/data_ingestion/resampling/timeseries_resampler.py

```diff
@@ -0,0 +1,118 @@
+"""
+TimeSeriesResampler class that delegates resampling to individual resmaplers.
+
+Copyright
+Copyright © 2022 Frequenz Energy-as-a-Service GmbH
+
+License
+MIT
+"""
+import logging
+from datetime import datetime
+from typing import Callable, Dict, Optional, Sequence
+
+import pytz
+
+from frequenz.sdk.core.timeseries import Sample, TimeSeriesId
+from frequenz.sdk.data_ingestion.resampling.individual_timeseries_resampler import (
+    IndividualTimeSeriesResampler,
+)
+
+logger = logging.Logger(__name__)
+
+ResamplingFunction = Callable[[Sequence[Sample], float], float]
+
+DEFAULT_BUFFER_SIZE = 100
+
+
+class TimeSeriesResampler:
+    """TimeSeriesResampler class that delegates resampling to individual resmaplers."""
+
+    def __init__(
+        self,
+        period_s: float,
+        max_data_age_in_periods: float,
+        resampling_function: ResamplingFunction,
```

Comment:
I would name the resampling function `initial_resampling_function`, as in the future it will change dynamically.

I also think (and this is a general comment) we should start using keyword-only arguments for stuff like `float`s, because they can be very easily mixed up and are hard to read from the caller side. And maybe add some defaults for stuff that make sense:

```suggestion
        self,
        *,
        period_s: float,
        max_data_age_in_periods: float = 3,
        initial_resampling_function: ResamplingFunction = average,
```

You would need to also define the `average` function of course.

Reply:
Ah, sorry, I see the defaults were included in the actor. Not sure now if both should have them, probably yes?

Reply:
Renamed the function and I agree with keyword-only arguments, good point! 👍 

As for defaults, I added a default for `max_data_age_in_periods`, but left the resampling function with a default value, because it is already defined in the `TimeseriesResamplingActor` (the average function is there).